### ISSUE TITLE
[question] [surface] What's the optimization algorithm used in projectPointOrthogonalToPolynomialSurface

### ISSUE THREAD

**Issue by keineahnung2345** [PointCloudLibrary/pcl#4913]
I am reading the code of `projectPointOrthogonalToPolynomialSurface` in `mls.hpp`, and I don't fully understand how it works. Following is my guess, please CMIIW: 
To find a closet point(gu,gv,gw) of `query_point`(u,v,w) on the surface, instead of optimizing their distance, this function calculates the partial derivative of their squared distance w.r.t `gu` and `gv`, getting F1 and F2, and try to minimizes them. 
The optimization algorithm is gradient descent(?), it calculates the Jacobian of F1,F2 w.r.t gu,gv, and finally calculates the update as `- J.inverse () * err`.
And here is my question: why does it calculates the update as `- J.inverse() * err`? In gradient descent, the update is calculated as negative gradient multiplied by step size/learning rate. So why `J.inverse()` not just `J` and why multiplied by `err`?
Thanks in advance.

https://github.com/PointCloudLibrary/pcl/blob/85fc1707dd28f4104a7cf6ab0f8570d11ec1b4c4/surface/include/pcl/surface/impl/mls.hpp#L586-L604

**Comment by mvieth** [comment]
I am not super familiar with MLS, but it seems to me that this is [Newton's method](https://en.wikipedia.org/wiki/Newton%27s_method). I assume you also read the doxygen documentation and saw [this stackexchange link](https://math.stackexchange.com/questions/1497093/shortest-distance-between-point-and-surface)?

**Comment by keineahnung2345** [comment]
In Newton's method, one needs to calculate Hessian matrix, so I'm afraid it's not the algorithm used here. And it doesn't look like Gauss-Newton or Levenberg-Marquardt because there is not something like J^T*J in the code.

> I assume you also read the doxygen documentation and saw this stackexchange link? 

Yes. I‘ve read them and [the stackexchange link](https://math.stackexchange.com/questions/1497093/shortest-distance-between-point-and-surface) makes me understand why it optimizes F1,F2 rather than the squared distance here. 

BTW, after some days, the code seems more reasonable to me, and following is my induction:
![Screenshot from 2021-09-05 18-56-01](https://user-images.githubusercontent.com/18047300/132124248-67df014f-2b0d-47c3-ad8c-5728ab6acfae.png)

But I still don't know what's the name of algorithm used here.

**Comment by mvieth** [comment]
> In Newton's method, one needs to calculate Hessian matrix, so I'm afraid it's not the algorithm used here

[Newton's method itself](https://en.wikipedia.org/wiki/Newton%27s_method) is for finding the root of a function f. It needs only the first derivative (Jacobian) of f. To minimize/maximize a function g, Newton's method can be applied to its first derivative g', in that case the second derivative of g (Hessian) is needed (see also [this Wikipedia article](https://en.wikipedia.org/wiki/Newton%27s_method_in_optimization)). My thought was that the former version is used here (the one for root finding, not the one for optimization)

**Comment by keineahnung2345** [comment]
> the one for root finding, not the one for optimization

The code seems to match Newton method for finding root perfectly, thank you!
